### DISCUSSION TITLE
create-nuxt-app prompt for nuxi init

### DISCUSSION THREAD

**Discussion by Baroshem** [nuxt/nuxt#17617] [Ideas]
Hey once again!

Today I would like to propose an addition to nuxi CLI that would enable users to scaffold an empty project with some integrations already installed and registered. Something like this was possible for Nuxt 2 using create-nuxt-app (https://github.com/nuxt/create-nuxt-app) and I remember that I was using it very frequently to automatically add CSS Frameworks, UI Libraries, features, etc.

Considering that the amount of integrations is rising (https://modules.nuxtjs.org/?version=3.x) it could be beneficial to update the nuxi cli `init` command to also use prompts to get the info from the user and generate adequate project.

I would propose prompts like following:

1. CSS Framework
* TailwindCSS
* WindiCSS
* UnoCSS
* none
2. UI Library
* Inkline
* In future more like Vuetify, PrimeVue, etc
* none
3. Features
* Pinia
* Harlem
* none
4. CMS
* Strapi
* Primsic
* Sanity
* Storyblok
* Directus
* none
5. Utils
* Color Mode
* Unlighthouse
* Partytown
* Plausible
* none

![image](https://user-images.githubusercontent.com/37120330/161046178-012a1bc0-c60c-4dfc-9ee7-68abc4647228.png)

This is just a proposed approach that can be changed to better suit the developers. 

I can create a Proof of Concept similarly to what I have created for `nuxi add` command

**Comment by Intevel** [comment]
A PoC would be great, with Nuxi we have a strong CLI tool, with a new command like `create-nuxt-app` from V2 it becomes even stronger.

**Comment by vinayakkulkarni** [comment]
Why not just refactor `create-nuxt-app` to spit out  Nuxt 2 as well as Nuxt 3 app?

**Reply by Baroshem** [reply]
I think for Nuxt 3 the plan is to go for nuxi with all CLI related functionalities.

**Reply by Intevel** [reply]
Yes, as far as I know that is the case. I think so, if we implement that as a new command in `Nuxi` that would be good.

**Comment by Baroshem** [comment]
What are your thoughts @pi0 @danielroe @Atinux ?

**Reply by atinux** [reply]
I highly recommend you to take a look at https://github.com/nuxt/framework/discussions/569

This is why we did not migrate `create-nuxt-app` to Nuxt 3.

**Reply by Baroshem** [reply]
I see. 

So If I understand correctly, the way here is to focus mainly on `nuxi module add @nuxtjs/color-mode` and not include templates at all (as proposed in my RFC)?

Can I help with developing that in this case?

**Reply by Diizzayy** [reply]
@Baroshem #3903 #3915

This would inherently allow us to offer more granular control over projects, as well as open the door to brand new ways of management or even initialization via Nuxt devtools ( powered by underlying CLI API service )

As it pertains to templating, A solution will be revised when we have a proper utility from @unjs

**Comment by pi0** [comment]
Thanks for the feedback and idea @Baroshem.

As @Atinux and @Diizzayy mentioned, the way we are moving forward with `nuxi` CLI and scaffolding is to add features progressively rather than a one-time setup. Often, users don't know what to start with and there is not as easy to just remove an option previously selected.

Once we had proper templating support, I don't see any reason why we cannot upgrade `create-nuxt-app` to use nuxi powered CLI and provide a wizard with steps and presets. Not judging `create-*` pattern is a bad idea that most the popular frameworks already have a CLI for. And more importantly, it can power devtools (or a browser wizard) to scaffold project with a specific preset for quick starts. (please consider all ideas as early ideas regarding `create-nuxt-app` future)

**Reply by katerlouis** [reply]
@pi0 Any news here? Or have I missed something and theres a completely new way of doing it? As of right now, I still install everything by hand after initilization.

**Comment by katerlouis** [comment]
Any new information on scaffolding / templating?

I am not very good at npm and setup stuff, so moving to Nuxt 3 without a similar experience then with `create-nuxt-app` in Nuxt 2 will be quite the challange.

When you say features like this get worked on progressively, is there a roadmap somewhere? Or a list on priorities? Adding linting on initialization for instance is arguably more important than lets say a framework like Tailwind.

**Comment by katerlouis** [comment]
For those like me who didn't know the cli got it's own repo: upvote here https://github.com/nuxt/cli/issues/53 to get the lovely wizard back <3
